app/controllers/users_controller.rb

```diff
@@ -40,10 +43,20 @@ def user_params
 
   def authorize
     authenticate_user!
-    if current_user.id.to_s == params[:id]
-      @user = current_user
-    else
-      redirect_to root_path
-    end
+    return unless params.key?(:id)
+
+    redirect_to users_path, alert: I18n.t('denial.not_found') unless User.exists?(params[:id])
+    update_user
+  end
+
+  def authorize_to_update!
+    return @user if current_user.id.to_s == params[:id]
+
+    redirect_to users_path, alert: I18n.t('denial.forbidden')
+    nil
```

Comment:
```suggestion
    redirect_to users_path, alert: I18n.t('denial.forbidden') unless current_user.id.to_s == params[:id]
```

Reply:
But in this case the return value will not be non-nil if `current_user.id.to_s == params[:id]` evaluates to `true`, will it?

Reply:
I think that it could work
```ruby
'test' unless 1 == 1
=> nil
'test' unless 1 == 2
=> "test"
```

Reply:
If `current_user.id.to_s == params[:id]` evaluates to `true`, the current implementation would return `@user` (not `nil`). But with your change, it would return `nil`, wouldn't it? In the other case, `redirect_to` will return a non-`nil` value instead of `nil`, so the logic is the opposite. Should it be this instead?

```rb
not (redirect_to users_path, alert: I18n.t('denial.forbidden') unless current_user.id.to_s == params[:id])
```

Reply:
That would be the better solution, but RuboCop will require `!` instead of `not`

Reply:
Ah, nevermind ... this diff has been outdated anyway.